src/main/java/duke/command/TaskOperation.java

```diff
@@ -0,0 +1,164 @@
+package duke.command;
+
+import duke.exception.DukeEmptyParaException;
+import duke.exception.DukeException;
+import duke.exception.DukeOutOfRangeException;
+import duke.task.Deadline;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.ToDos;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskOperation {
+    // Attributes
+    private final ArrayList<Task> tasks; // using Java Collections classes
+
+    public TaskOperation(){
+        tasks = new ArrayList<>();
+    }
+
+    public void doneTask(String userInput) throws DukeException {
+        int i = userInput.indexOf(" ");
+        if(i == -1){
+            throw new DukeEmptyParaException("There should be an index of task in the done :-(");
+        }
+
+        int taskIndex = Integer.parseInt(userInput.substring(i + 1));
+
+        if(taskIndex > tasks.size()){
+            throw new DukeOutOfRangeException("Please enter a valid index of task :-(");
+        }
+
+        tasks.get(taskIndex - 1).markAsDone();
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(tasks.get(taskIndex - 1));
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+
+    }
+
+    public void addTask(String command, String userInput) throws DukeException {
+        int i = userInput.indexOf(" ");
+        String taskInfo = " ";
+        if(i != -1){
+            taskInfo = userInput.substring(i + 1);
+        }
+
+        switch (command) {
+            case "todo":
+                tasks.add(ToDos.parse(taskInfo));
+                break;
+            case "deadline":
+                tasks.add(Deadline.parse(taskInfo));
+                break;
+            case "event":
+                tasks.add(Events.parse(taskInfo));
+                break;
+        }
+
+        System.out.println(Duke.MESSAGE_BOUNDARY + "\n" + "Got it. I've added this task: ");
+        System.out.println(tasks.get(tasks.size() - 1));
```

Comment:
Including such nested logic inside the print statement itself makes the code hard to read. Better to separate out the logic as follows:
int taskSize = task.size();
String lastTask = tasks.get(taskSize - 1);
System.out.println(lastTask);